lib/Conversion/TorchToTcp/Misc.cpp

```diff
@@ -133,6 +133,39 @@ class ConvertValueTensorLiteralOp
   }
 };
 
+class ConvertAtenSizeIntOp : public OpConversionPattern<AtenSizeIntOp> {
+public:
+  using OpConversionPattern<AtenSizeIntOp>::OpConversionPattern;
+
+  LogicalResult
+  matchAndRewrite(AtenSizeIntOp op, OpAdaptor adaptor,
+                  ConversionPatternRewriter &rewriter) const override {
+    Location loc = op->getLoc();
+    Value self = adaptor.getSelf();
+    auto type = self.getType().cast<RankedTensorType>();
+    if (!isa<ConstantIntOp>(op->getOperand(1).getDefiningOp())) {
+      return rewriter.notifyMatchFailure(op, "dim must be a constant int");
+    }
+    auto constIntOp =
+        dyn_cast<ConstantIntOp>(op->getOperand(1).getDefiningOp());
+    int idxVal = constIntOp.getValueAttr().getValue().getSExtValue();
+    if (idxVal < 0 || idxVal >= type.getRank()) {
+      return rewriter.notifyMatchFailure(op, "dim must be in range");
+    }
+    auto idxOp = rewriter.create<arith::ConstantIndexOp>(loc, idxVal);
+    auto dimOp = rewriter.create<tensor::DimOp>(loc, self, idxOp);
+    auto result =
+        rewriter.create<arith::IndexCastOp>(loc, rewriter.getI64Type(), dimOp);
+
+    rewriter.replaceOp(op, result);
+    if (constIntOp->hasOneUse()) {
+      rewriter.eraseOp(constIntOp);
+    }
```

Comment:
Why is this necessary? Shouldn't this be automatically taken care of by DCE pass (canonicalizer)?

Reply:
Good point. Just removed.